code/src/main/kotlin/lexer/Tokens.kt

```diff
@@ -0,0 +1,34 @@
+package lexer
+
+sealed class Token(var lineNumber: Int? = null, var lineIndex: Int? = null) {
+    class Identifier(val value: String) : Token()
+    sealed class Literal : Token()  {
+        class Text(val value: String) : Literal()
+        class Number(strValue: String, val value: Double = strValue.toDouble()) : Literal()
+        class Bool(strValue: String, val value: Boolean = strValue.toBoolean()) : Literal()
+    }
+    sealed class SpecialChar : Token() {
+        class BlockStart : SpecialChar()
+        class BlockEnd : SpecialChar()
+        class SquareBracketStart : SpecialChar()
+        class SquareBracketEnd : SpecialChar()
+        class ParenthesesStart : SpecialChar()
+        class ParenthesesEnd : SpecialChar()
+        class EndOfLine : SpecialChar()
+        class ListSeparator : SpecialChar()
+        class LineContinue : SpecialChar()
+        class Equals : SpecialChar()
+        class Arrow : SpecialChar()
+    }
+    sealed class Type : Token() {
+        class Var : Type()
+        class None : Type()
+        class Text : Type()
+        class Number : Type()
+        class Bool : Type()
+        class Tuple : Type()
+        class List : Type()
+        class Func : Type()
+    }
```

Comment:
Just a minor question. In the EBNF the "dict" type is included in types, is there a reason why it is not a sublclass in the Type class.

Reply:
I realize it is a nice to have feature, but the EBNF and the Type class should reflect eachother

Reply:
i don't think dict needs to be in the EBNF as per our conversations, but i would love to hear everyone elses opinions

Reply:
I think it should be removed from the EBNF. It doesn't seem important to our goal, and I don't think we will learn much from including it, so it's just more work for no reason.